### PR TITLE
More fixes for ai_dynamic_skill, optimizations and working warnings

### PR DESCRIPTION
**Pull request by freghar**
See commit messages for details, ... this should be the final batch of fixes, hopefully.

Note that warnings may appear if `CustomAILevel` is not set correctly according to the README, whereas in the previous (current master) version, a bug (copy/paste from legacy code) prevented them from showing up.

### CODE REVIEW

File: addons/ai_dynamic_skill/fn_setSkills.sqf [hunk 1]
```diff
@@ -1,40 +1,45 @@
...
+             * set > 0:    because we cannot distinguish between createVehicle and
+             *             createUnit, the latter having initially skill/skillFinal 0
+             * abs check:  because of floating point MP errors in the final skill
+             */
+            if (_val >= 0.2 && _skill > 0 && abs (_val-_final) > 0.05) then {
+                private _msg = format [
+                    "ai_dynamic_skill: skill %1 is %2, but skillFinal is %3",
+                    _name, _val toFixed 2, _final toFixed 2
+                ];
+                _msg remoteExec ["systemChat"];
```
Comment (milivojm) [inline addons/ai_dynamic_skill/fn_setSkills.sqf:39]: Do we ever change `precisionAI` and `skillAI` on the fly?

**Review by lastmikoi — APPROVED** [review]

**Comment by lastmikoi** [comment]
Adding @Abuk007 to the reviewers. Will land in tomorrow's AI test if he approves it.

**Comment by milivojm** [comment]
Looks very nice. Thank you @freghar !

File: addons/ai_dynamic_skill/fn_setSkills.sqf (hunk 1 above)
Comment (freghar) [reply addons/ai_dynamic_skill/fn_setSkills.sqf:39]: No, and it's a good point that I could just check for them once in a `preStart` function (on game start) and print warnings to RPT, but - from experience - server admins never check those and this setting is vital for proper skill settings.

Hopefully, after the first 10 minutes of game runtime, the overhead should reduce to virtually nothing.

**Comment by Abuk007** [comment]
Approved guys, get it rolling ASAP.

**Pull request merged**